tests/integration/models/test_well_models.py

```diff
@@ -0,0 +1,415 @@
+            subdomains: List of subdomains for which to define boundary conditions.
+
+        Returns:
+            bc: Boundary condition object.
+
+        """
+        val = self.fluid.convert_units(-1, "kg * m ^ 3 * s ^ -1")
+        return self._bc_values(subdomains, val, "bc_values_darcy")
+
+    def bc_type_mobrho(self, sd: pp.Grid) -> pp.BoundaryCondition:
+        return self._bc_type(sd, "dir")
+
+    def bc_values_mobrho(self, subdomains: list[pp.Grid]) -> pp.ad.DenseArray:
+        """Boundary condition values for Darcy flux.
+
+        Dirichlet boundary conditions are defined on the north and south boundaries,
+        with a constant value of 0 unless fluid's reference pressure is changed.
+
+        Parameters:
+            subdomains: List of subdomains for which to define boundary conditions.
+
+        Returns:
+            bc: Boundary condition object.
+
+        """
+        val = self.fluid.density()
+        return self._bc_values(subdomains, val, "bc_values_mobrho")
+
+    def bc_type_enthalpy(self, sd: pp.Grid) -> pp.BoundaryCondition:
+        """Boundary condition type for enthalpy.
+
+        Dirichlet boundary conditions are defined on the north and south boundaries.
+
+        Parameters:
+            sd: Subdomain for which to define boundary conditions.
+
+        Returns:
+            bc: Boundary condition object.
+
+        """
+        return self._bc_type(sd, "dir")
+
+    def bc_values_enthalpy_flux(self, subdomains: list[pp.Grid]) -> pp.ad.DenseArray:
+        """Boundary condition values for enthalpy.
+
+        Dirichlet boundary conditions are defined on the north and south boundaries,
+        with a constant value of 0 unless fluid's reference pressure is changed.
+
+        Parameters:
+            subdomains: List of subdomains for which to define boundary conditions.
+
+        Returns:
+            bc: Boundary condition object.
+
+        """
+        val = self.fluid.convert_units(
+            1e7, "kg * m ^ 3 * s ^ -1"
+        )  # * self.fluid.density() * self.fluid.specific_heat_capacity()
+        return self._bc_values(subdomains, val, "bc_values_enthalpy")
+
+    def bc_type_fourier(self, sd: pp.Grid) -> pp.BoundaryCondition:
+        """Boundary condition type for Fourier flux.
+
+        Dirichlet boundary conditions are defined on the north and south boundaries.
+
+        Parameters:
+            sd: Subdomain for which to define boundary conditions.
+
+        Returns:
+            bc: Boundary condition object.
+
+        """
+        return self._bc_type(sd, "neu")
+
+
+class TailoredPermeability(pp.constitutive_laws.CubicLawPermeability):
+    def permeability(self, subdomains: list[pp.Grid]) -> pp.ad.Operator:
+        """Permeability [m^2].
+
+        This function combines a matrix permeability with a cubic law permeability for
+        fractures and intersections. The combination entails projection between the two
+        subdomain subsets and all subdomains.
```

Comment:
Please add a one-sentence motivation. If I ever need to debug this function, I will be extremely grateful.